### DISCUSSION TITLE
Mismatches detected

### DISCUSSION THREAD

**Discussion by abaibaihaqi** [mifi/lossless-cut#2041] [General]
How to fix this  

Track 1 mismatch: codec_name h264 != aac
Track 1 mismatch: width 720 != none
Track 1 mismatch: height 1392 != none
Track 1 mismatch: pix_fmt yuv420p != none
Track 1 mismatch: level 40 != none
Track 1 mismatch: profile High != LC
Track 1 mismatch: sample_fmt none != fltp
Track 1 mismatch: r_frame_rate 45000/1 != 0/0
Track 1 mismatch: time_base 1/90000 != 1/44100
Track 2 mismatch: codec_name aac != h264
Track 2 mismatch: width none != 720
Track 2 mismatch: height none != 1392
Track 2 mismatch: pix_fmt none != yuv420p
Track 2 mismatch: level none != 40
Track 2 mismatch: profile LC != High
Track 2 mismatch: sample_fmt fltp != none
Track 2 mismatch: r_frame_rate 0/0 != 45000/1
Track 2 mismatch: time_base 1/44100 != 1/90000

**Comment by mifi** [comment]
It seems like your tracks have become flipped in one of your files. you could extract each stream as its own file, and then combine them together again (dropping each extracted file into losslesscut in the correct order)

**Reply by dreeddoes** [reply]
Hey, I have the same error:

Mismateches detected

Track 1 mismatch: codec_name aac != h264
Track 1 mismatch: width none != 1600
Track 1 mismatch: height none != 900
Track 1 mismatch: pix_fmt none != yuv420p
Track 1 mismatch: level none != 42
Track 1 mismatch: profile LC != High
Track 1 mismatch: sample_fmt fltp != none
Track 1 mismatch: r_frame_rate 0/0 != 60/1
Track 1 mismatch: time_base 1/48000 != 1/90000
Track 2 mismatch: codec_name h264 != aac
Track 2 mismatch: width 1600 != none
Track 2 mismatch: height 900 != none
Track 2 mismatch: pix_fmt yuv420p != none
Track 2 mismatch: level 42 != none
Track 2 mismatch: profile High != LC
Track 2 mismatch: sample_fmt none != fltp
Track 2 mismatch: r_frame_rate 60/1 != 0/0
Track 2 mismatch: time_base 1/90000 != 1/48000

I tried to understand what you mean but I don't know what to extract and then what to do. Can you maybe make a bit more comprehensive guide on that to do? I have other 2 files that have the exact same problem. It would really help to fix that.

**Comment by MickeeeM** [comment]
I'm having the exact problem. I have lots of mpg files from an old camera that I want to merge together into one video. Most of them are fine, but for some reason few of them come up with this mismatch message where it seems everything is flipped. Therefor not able to merge those.

Anyone know a solution? I'm not good at these things as I just downloaded Losslesscut to merge files together.

**Reply by mifi** [reply]
Did you try my above suggestion by extracting streams as files and then combining them into a single file in the correct order?

**Reply by fy860825** [reply]
@mifi Can you share the steps to perform the stream extration as files and then combine into single file? 
Pretty unsure how to do this...

**Reply by mifi** [reply]
ok, assuming your file has these two tracks, in the following order:
1. audio
2. video

...and assuming you want them to be flipped (video first, then audio), here's what I would do:

1. open file
2. open Tracks panel (top right)
3. disable the audio track (track 1)
4. for the video track (track 2), on the right side of the row, click the three dots, then "Extract this track as file"
5. click "include more tracks from other file"
6. choose the file you just extracted in step 4
7. close tracks panel
8. export

**Reply by fy860825** [reply]
@mifi I have perform the steps but seems more mismatch parameter appeared. 

Probably I'll explain slighly more, I use capcut to export my mp4 files all in same format, same parameters, but each video has different in length. When I try to merge them I get this message.
Track 1 mismatch: time_base 1/30 != 1/15360

If I follow above steps I seems to receive more mismatch, unsure if I follow the steps right.
Track 1 mismatch: codec_name aac != h264
Track 1 mismatch: width none != 1920
Track 1 mismatch: height none != 1080
Track 1 mismatch: pix_fmt none != yuv420p
Track 1 mismatch: level none != 40
Track 1 mismatch: profile LC != Main
Track 1 mismatch: sample_fmt fltp != none
Track 1 mismatch: r_frame_rate 0/0 != 30/1
Track 1 mismatch: time_base 1/44100 != 1/15360

Some other project I work on does not have problem, same capcut export, same parameters...

**Reply by fy860825** [reply]
Some other video which give slightly different in mismatch
Track 1 mismatch: width 1280 != 1920
Track 1 mismatch: height 720 != 1080
Track 1 mismatch: level 31 != 40
Track 1 mismatch: time_base 1/30 != 1/15360

**Comment by fy860825** [comment]
I observe the files having diferences below even they have same parameters and encode during their export from capcut

  file 1
  time_base: '1/15360',
  start_pts: 0,
  start_time: '0.000000',
  duration_ts: 159702016,
  duration: '10397.266667',
  bit_rate: '2316453',
  bits_per_raw_sample: '8',
  nb_frames: '311918',
  
  file 2
  time_base: '1/30',
  start_pts: 0,
  start_time: '0.000000',
  duration_ts: 304961,
  duration: '10165.366667',
  bit_rate: '2322593',
  bits_per_raw_sample: '8',
  nb_frames: '304961',

**Reply by mifi** [reply]
if you want to normalise timebase, you can try to remux each file to `mpegts` first before merging, see https://github.com/mifi/lossless-cut/blob/master/issues.md#merge--concat-results-in-corrupt-or-broken-parts

**Reply by fy860825** [reply]
thanks. i've got chatgpt to write me some simple code to make use ffmpeg to fix the mismatch and merging works.
